x-pack/platform/plugins/shared/onechat/public/application/components/tools/esql/esql_tool.tsx

```diff
@@ -66,7 +69,9 @@ export const EsqlTool: React.FC<EsqlToolProps> = ({
   const { navigateToOnechatUrl } = useNavigation();
   const form = useEsqlToolForm();
   const { reset, formState } = form;
-  const { errors } = formState;
+  const { errors, isDirty } = formState;
+  const [showTestFlyout, setShowTestFlyout] = useState(false);
+  const [testToolData, setTestToolData] = useState<OnechatEsqlToolFormData>(form.getValues());
```

Comment:
Why do we need this state? Can't we infer `testToolData` directly from `const form = useEsqlToolForm();`?

Reply:
We are always forcing user to save before executing the tool, we could even simplify to pass just toolId to flyout test tool form , and then from within flyout form we can fetch the tool by Id using a hook `useTool(toolId)`

IMO this will be cleaner and we won't need to handle state propagation to child component